src/pages/resultsView/ResultsViewPageStoreUtils.ts

```diff
@@ -572,40 +572,45 @@ export function makeEnrichmentDataPromise<
         qValue?: number;
     }
 >(params: {
-    storeForExcludingQueryGenes?: ResultsViewPageStore;
+    resultsViewPageStore?: ResultsViewPageStore;
     await: MobxPromise_await;
     referenceGenesPromise: MobxPromise<{
         [hugoGeneSymbol: string]: ReferenceGenomeGene;
     }>;
-    getSelectedProfileMap: () => { [studyId: string]: MolecularProfile };
+    getSelectedProfileMaps: () => { [studyId: string]: MolecularProfile }[];
     fetchData: () => Promise<T[]>;
 }): MobxPromise<(T & { qValue: number })[]> {
     return remoteData({
         await: () => {
             const ret = params.await();
-            if (params.storeForExcludingQueryGenes) {
-                ret.push(
-                    params.storeForExcludingQueryGenes.selectedMolecularProfiles
-                );
+            if (params.resultsViewPageStore) {
+                ret.push(params.resultsViewPageStore.selectedMolecularProfiles);
             }
             ret.push(params.referenceGenesPromise);
             return ret;
         },
         invoke: async () => {
-            const profileMap = params.getSelectedProfileMap();
-            if (profileMap) {
+            const profileMaps = params.getSelectedProfileMaps();
+            if (profileMaps) {
                 let data = await params.fetchData();
                 // filter out query genes, if looking at a queried profile
                 // its important that we filter out *before* calculating Q values
-                if (
-                    params.storeForExcludingQueryGenes &&
-                    params.storeForExcludingQueryGenes.selectedMolecularProfiles.result!.findIndex(
-                        molecularProfile =>
-                            profileMap[molecularProfile.studyId] !== undefined
-                    ) > -1
-                ) {
+                const filterQueryGenes =
```

Comment:
make this existsFilterQueryGenes since it's boolean

Reply:
Would you agree with `doFilterQueryGenes`? It is the need for filtering that is symbolized by the boolean.